runtime/util/hshelp.c

```diff
@@ -3171,6 +3171,48 @@ verifyMethodsAreSame(J9VMThread * currentThread, J9JVMTIClassPair * classPair, U
 
 }
 
+#if (JAVA_SPEC_VERSION >= 14)
+static jvmtiError
+verifyRecordAttributesAreSame(J9ROMClass *originalROMClass, J9ROMClass *replacementROMClass)
+{
+	jvmtiError rc = JVMTI_ERROR_NONE;
+
+	/* Since retranformation is not allowed to change inheritance there's no need to consider 
+	 * one class being a record and one not. */
+	if (J9ROMCLASS_IS_RECORD(originalROMClass) && J9ROMCLASS_IS_RECORD(replacementROMClass)) {
+		U_32 originalNumberOfRecords = getNumberOfRecordComponents(originalROMClass);
+		U_32 replacementNumberOfRecords = getNumberOfRecordComponents(replacementROMClass);
+
+		if (originalNumberOfRecords == replacementNumberOfRecords) {
+			if (originalNumberOfRecords > 0) {
+				J9ROMRecordComponentShape* originalRecordComponent = NULL;
+				J9ROMRecordComponentShape* replacementRecordComponent = NULL;
+				U_32 i = 0;
+
+				originalRecordComponent = recordComponentStartDo(originalROMClass);
+				replacementRecordComponent = recordComponentStartDo(replacementROMClass);
```

Comment:
This assumes that record components can't be reordered.  Is there a connonical order defined in the spec?  If not, how does the RI handle this?  Does it require that the order of the components is stable?

Reply:
The spec requires that the order of record components be the same as the order in which they are declared. Since retransformation is not allowed to change fields the record components should be expected to maintain the same order.

Reply:
Sorry to unresolve this.  Can you add a comment to that effect into the code?  It'll save rediscovering that in the future

Reply:
yes. good idea.